packages/dds/tree/docs/data-model/README.md

```diff
@@ -0,0 +1,166 @@
+From the data model's perspective, field keys are opaque.
+
+### Sequences
+
+In the SharedTree data model abstraction, there is no distinction between a field that always contain a single child and a field that can contain multiple children.
+Instead, SharedTree treats all fields as ordered collections and uses schema to restrict which fields are optional, must contain a single value, or may contain multiple values.
+
+When mapping the SharedTree data model to conventional programming languages, it is sometimes helpful to think of the sequence as being distinct from the field, as if each field points to an ***implicit sequence*** object.
+
+However, it is important to remember that sequences are not tree nodes.
+This means that sequences are not directly addressable.
+The combination of the field + sequences is implicitly created when the first item is inserted, implicitly deleted when the last item is removed, and may only be referenced indirectly via the combination of parent node + field key.
+
+### Special Fields
+
+This section covers fields that receive special treatment in the SharedTree data model.
+These fields are special because:
+
+- They have well-known keys
+- They are universally available on all nodes (regardless of schema.)
+- They can not be targeted by normal tree operations.
+
+#### Type
+
+The SharedTree data model supports nominal typing via a special *type* field.
+The value of the *type* field is the unique identifier of the corresponding schema type.
+
+In the data model abstraction, the value of the type field is opaque.
+However, there is a set of well-known types (*boolean*, *number*, etc.) that are transparent to the underlying implementation.
+
+#### Id
+
+The SharedTree data model supports durable references to tree nodes via a special *id* field.
+The *id* field is used by SharedTree to provide a bidirectional *id* ⟷ *node* index that provides efficient lookup of nodes by id.
+This is the underlying building block for features like "share link" URLs and graph-like references within the tree.
+
+## JSON Comparison
+
+The below diagram highlights the differences between the JSON data model and the SharedTree data model using the following snippet:
+
+```json
+{
+    "visible": true,
+    "text": "cat",
+    "dashStyle": [0.5, 3]
+}
+```
+
+<figure align="center">
+  <img src="./assets/data-model.drawio.svg" alt=""/>
+  <figcaption>Figure: JSON and SharedTree</figcaption>
+</figure>
+
+Of note:
+
+- Scalar values are represented by tree nodes and consequently have a durable identity.
+- The 'visible' field is a sequence, even though it is constrained by schema to only contain a single boolean value.
+- The 'text' field leverages it's implicit sequence to represent the letters of "cat" as individual nodes, allowing the text to be collaboratively edited in the same way as array.
+- Unlike the 'text' field, an explicit tree node is used to represent the JavaScript array object.
```

Comment:
I think this sentence forgot to mention that it is talking about the `dashStyle` field.

Should this say `JSON` instead of `JavaScript` as well?

Reply:
Changed to just 'the array object', thx.